docs/NNNN-direct-clean-path-layout.md

```diff
@@ -0,0 +1,451 @@
+# OCFL Community Extension NNNN: Direct Clean Path Layout
+
+* **Extension Name:** NNNN-direct-clean-path-layout
+* **Authors:** Jürgen Enge (Basel)
+* **Minimum OCFL Version:** 1.0
+* **OCFL Community Extensions Version:** 1.0
+* **Obsoletes:** n/a
+* **Obsoleted by:** n/a
+
+## Overview
+
+This extension can be used to either as a storage layout extension that maps OCFL 
+object identifiers to storage paths or as an object extension that maps logical paths to content paths.
+This is done by replacing or removing "dangerous characters" from names.
+
+This functionality is based on David A. Wheeler's essay "Fixing Unix/Linux/POSIX Filenames" (https://www.dwheeler.com/essays/fixing-unix-linux-filenames.html)
+
+### Usage Scenario
+
+This extension is intended to be used when you want the internal OCFL layout to match 
+the logical structure of the stored objects as closely as possible, while also 
+ensuring all file and directory names do not include problematic characters.
+
+One example could be complete filesystems, which come from estates, and are not
+under control of the archivists.
+
+### Caveat
+
+#### Projection
+This extension provides injective (one-to-one) projections only if 
+`encodeUTF == true`.  
+If you don't want to use the verbose UTF-Code replacements (`encodeUTF == false`), first check your source to make sure, that there's no chance of 
+different names to be mapped on one target. This happens on whitespace replacement as well as on 
+the removal of leading `-`, `~` and ` ` (blank) or trailing ` ` (blank)
+Software which generates the OCFL 
+structure should raise an error in this case. 
+##### Example names, that would map to the same target
+* `~file` => `file`
+* `-file` => `file`
+* ` file` => `file`
+* `file` => `file`
+* `file ` => `file`
+
+#### Object Identifiers
+When using this extension as a storage layout, you must ensure that none of your 
+object identifiers are prefixes of other identifiers. For example, 
+`https://hdl.handle.net/XXXXX/test` and `https://hdl.handle.net/XXXXX/test/blah`. 
+This is a problem because it would result in the storage paths 
+`https/hdl.handle.net/XXXXX/test` and `https/hdl.handle.net/XXXXX/test/blah`, which is 
+invalid because the first object contains the second.
+
+
+## Parameters
+
+### Summary
+
+* **Name:** `encodeUTF`
+    * **Description:** Decides whether "dangerous" characters will be replaced by a 
+      defined replacement string or it's utf code (e.g. `=u0020` for blank char) 
+    * **Type:** bool
+    * **Default:** false
+* **Name:** `maxFilenameLen`
```

Comment:
Would it be more clear to change this parameter name to something like: `maxPathSegmentLen` or `maxPathPartLen`?

Reply:
yes, changed to maxPathSegmentLen